### PR TITLE
ISSUE-967: Update Oracle Db Adapter to v4

### PR DESCRIPTION
**Pull request by BeMySlaveDarlin**
Issue: [#967](https://github.com/phalcon/incubator/issues/967)
- [x] Removed:
    `Phalcon\Db\Adapter\Cacheable\Mysql`
    `Phalcon\Db\Adapter\Mongo\Client`
    `Phalcon\Db\Adapter\Mongo\Collection`
    `Phalcon\Db\Adapter\Mongo\Cursor`
    `Phalcon\Db\Adapter\Mongo\Db`
    `Phalcon\Db\Adapter\Mongo\DbRef`
    `Phalcon\Db\Adapter\Mongo\Document`
    `Phalcon\Db\Adapter\MongoDB\Exception`
    `Phalcon\Db\Adapter\MongoDB\GridFS`
    `Phalcon\Db\Adapter\MongoDB\Model`
    `Phalcon\Db\Adapter\MongoDB\Operation`
    `Phalcon\Db\Adapter\MongoDB\*`
    `Phalcon\Db\Adapter\Factory`
    `Phalcon\Db\Dialect\MysqlExtended`
    `Phalcon\Db\Result\Serializable`
- [x] Updated to v4:
    `Phalcon\Db\Adapter\Pdo\Oracle`
    `Phalcon\Db\Dialect\Oracle`
- [x] Added Github workflow

### CODE REVIEW

**Comment by Jeckerson** [comment]
Namespace must be `Phalcon\Incubator\Db` to respect repository name and also to not have collision with Phalcon itself.

**Comment by BeMySlaveDarlin** [comment]
> Namespace must be `Phalcon\Incubator\Db` to respect repository name and also to not have collision with Phalcon itself.

Done

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    "phalcon/ide-stubs": "^4.0",
+    "vimeo/psalm": "^3.6",
+    "squizlabs/php_codesniffer": "^3.5",
+    "codeception/codeception": "^4.1",
+    "codeception/module-asserts": "^1.0.0",
+    "phpstan/phpstan": "^0.12.18"
+  },
+  "autoload": {
+    "psr-4": {
+      "Phalcon\\Db\\": "src/"
```
Comment (Jeckerson) [inline composer.json]: Fix namespace

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    }
+  },
+  "autoload-dev": {
+    "psr-4": {
+      "Phalcon\\Db\\Tests\\Unit\\": "tests/unit/"
```
Comment (Jeckerson) [inline composer.json]: Fix namespace

File: src/Adapter/Pdo/Oracle.php [hunk 3]
```diff
@@ -108,47 +107,47 @@ public function describeColumns($table, $schema = null)
          * 7:default,
          * 8:position;
          */
-        $sql = $this->_dialect->describeColumns($table, $schema);
-        foreach ($this->fetchAll($sql, Db::FETCH_NUM) as $field) {
-            $definition      = ['bindType' => 2];
-            $columnSize      = $field[2];
+        $sql = $this->dialect->describeColumns($table, $schema);
+        foreach ($this->fetchAll($sql, Enum::FETCH_NUM) as $field) {
```
Comment (Jeckerson) [inline src/Adapter/Pdo/Oracle.php]: Move outside cycle, it will execute every iteration.

**Review by Jeckerson — CHANGES_REQUESTED** [review]

**Review by Jeckerson — APPROVED** [review]

**Pull request merged**